tests/integration_tests/clouds.py

```diff
@@ -309,3 +318,27 @@ def _get_or_set_profile_list(self, release):
         self._profile_list = self.cloud_instance.build_necessary_profiles(
             release)
         return self._profile_list
+
+
+class OpenstackCloud(IntegrationCloud):
+    datasource = 'openstack'
+    integration_instance_cls = IntegrationInstance
+
+    def _get_cloud_instance(self):
+        return Openstack(
+            tag='openstack-integration-test',
+            network=integration_settings.OPENSTACK_NETWORK,
+        )
+
+    def _get_initial_image(self):
+        image = ImageSpecification.from_os_image()
+        try:
+            UUID(image.image_id)
+        except ValueError as e:
+            raise Exception(
+                'When using Openstack, `OS_IMAGE` parameter MUST take the '
+                'form of a 36-character UUID. Passing in a release name is '
```

Comment:
OS_IMAGE can still take its `::` separated form (and arguably _should_, given we can't possibly infer OS/release at the moment), it's just the first component of that form which is constrained in this way. Something along the lines of "OS_IMAGE MUST be specified with a 36-character UUID image ID", perhaps?

Reply:
This still needs addressing, I think?